packages/experimental/Avatar/SPEC.md

```diff
@@ -0,0 +1,311 @@
+- Brand - Brand colors from the theme
+- Colorful - Pick from a list of pre-defined Avatar colors. The color will be assigned based on a hash of the name (to "randomly" assign a person a color). The color name (like darkRed) can also be specified explicitly in case the use case requires a different algorithm to pick the color.
+
+### Icon
+
+The `Avatar` component can include an `icon` that appears if there's no image or initials.
+
+### Image
+
+The `Avatar` component can include an `image`.
+
+### Initials
+
+Initials can be passed using `initials` prop or generated from `name` prop using getInitials method.
+Here are the requirements from Web spec:
+
+1. If only one word is provided, pull the first letter of the word.
+2. If 2 words provided, pull first letter of both words.
+3. If 2+ words provided, pull first letter of first word and last word.
+4. Ignore suffixes (i.e. Jr, III).
+5. Ignore words in parantheses Ignore special characters i.e. parantheses, dashes If numeric, fall back to Icon If first character is not alphabetic, fall back to icon.
+
+But requirements are still being discussed. This area will be updated once we come up with final solition. There are different implementations in different teams.
+
+### Name
+
+Name prop is used to generate initials.
+
+### Shape
+
+The Avatar supports a circular and square (with rounded corners) shape.
+
+### Sizes
+
+`Avatar` sopports next sizes: `20, 24, 28, 32, 36, 40, 48, 56, 64, 72, 96, 120`.
+Default size for win32 is 24.
+
+## API
+
+### Slots
+
+The `Avatar` component has the following slots:
+
+- `root` - The outer container representing the `Avatar` wrapper.
+- `image` - If specified, renders an `image`
+- `initials` - If specified, renders initials when there's no `image`
+- `icon` - If specified, renders an `icon` if there's no `image` or `initials`.
+- `initialsBackground` - It's a wrapper for initials and icon.
+- `ring` - If specified, renders the ring for the `Avatar` when active prop is set to `active`.
+- `badge` - If specified, renders presence badge. Example: `badge={{ status: 'outOfOffice' }}.`.
+
+The slots can be modified using the `compose` function on the `Avatar`. For more information on using the `compose` API, please see [this page](../../framework/composition/README.md).
+
+### Props
+
+```ts
+export interface JSAvatarProps extends IViewProps, AvatarConfigurableProps {
+  /**
+   * Optional activity indicator
+   * * active: the avatar will be decorated according to activeAppearance
+   * * inactive: the avatar will be reduced in size and partially transparent
+   * * unset: normal display
+   *
+   * @defaultvalue unset
+   */
+  active?: AvatarActive;
+
+  /**
+   * The appearance when `active="active"`
+   *
+   * @defaultvalue ring
+   */
+  activeAppearance?: AvatarActiveAppearance;
+
+  /**
+   * Badge shows the avatar's presence status.
+   * Badge can be shown only if `active` prop is undefined or unset
+   */
+  badge?: PresenceBadgeProps;
```

Comment:
I would have guessed that `badge` and `icon` would be slots and not exposed directly as props. Did that change?

Reply:
There are slots. But we pass badge and icon parameters like this: `<JSAvatar badge={{ status: 'available'}} icon={{ fontSource: { ...fontBuiltInProps }, color: 'white' }} />`